### ISSUE TITLE
Segment fault starting component

### ISSUE THREAD

**Issue by krinehart** [hunterjm/hassio-addons#32]
HA 0.88.1 using Hassio component. When I try to start the component I get:
/run.sh: line 14:    12 Segmentation fault      (core dumped) xbox-rest-server

Not sure what else to try - no other information that I can find.

**Comment by jesryn** [comment]
Same for me

**Comment by unknownskl** [comment]
Which architecture is the server running on? (Amd64, arm, ..?)

**Comment by krinehart** [comment]
64bit Raspberry Pi 3 B+
Here's the System Health screen from Home Assistant:
arch  | aarch64
dev | false
docker | true
hassio | true
os_name | Linux
python_version | 3.7.2
version | 0.88.1
virtualenv | false

**Comment by jesryn** [comment]
System Health from me as well

arch | aarch64
-- | --
dev | false
docker | true
hassio | true
os_name | Linux
python_version | 3.7.2
timezone | Europe/Stockholm
version | 0.88.2
virtualenv | false

(Updated to 88.2 today)

**Comment by unknownskl** [comment]
I'm sure this has to do with the docker image that's build for amd64 and not for aarch64.
@hunterjm: I have an docker file that works in my fork from your repo. Maybe you can use that dockerfile? (Works on both aarch64 and amd64, however I don't use HA, as I've migrated to homebridge, so I can't test and make a pull request)

**Comment by jesryn** [comment]
@krinehart have you found any solution? Feels like I have to move on to something else for the Xbox integration now

**Comment by krinehart** [comment]
No, @jesryn, I'm holding out hope that @hunterjm can create a docker image for aarch64 as it sounds like that's the problem but I haven't heard anything.

**Comment by cbuechel** [comment]
Have the same issue. Hoping for a resolution

**Comment by jesryn** [comment]
I reinstalled and moved to the 32-bit version today. But when trying to reinstall the Xbox component it seems to have a general problem right now. Not possible to install from that repo. Found others with the same problem. But I will stick to the 32-bit Hassio a couple of days to see if the repo problem will be solved, otherwise might move back to 64-bit.

**Comment by webdjoe** [comment]
@hunterjm is probably very busy as usual.  Until he updates the repository a workaround might be to just use the fix from @unknownskl repository by using his fork path [https://github.com/unknownskl/hassio-addons](https://github.com/unknownskl/hassio-addons) to install a third party addon instead of this one.  I don't use hassio so I can't tell if it works but it looks alright to me.  Let me know if it is in fact successful.  If it is I will fork and make a pull request.

**Comment by jesryn** [comment]
> @hunterjm is probably very busy as usual. Until he updates the repository a workaround might be to just use the fix from @unknownskl repository by using his fork path https://github.com/unknownskl/hassio-addons to install a third party addon instead of this one. I don't use hassio so I can't tell if it works but it looks alright to me. Let me know if it is in fact successful. If it is I will fork and make a pull request.

Thanks for the info @webdjoe, appreciated. Will switch to your suggestion during the weekend.

**Comment by jampez77** [comment]
I've been having this solution for a while now. Has anyone found a solution to this? @jesryn did the @unknownskl idea work?

**Comment by SBRK** [comment]
Same here...

**Comment by unknownskl** [comment]
@SBRK it is probably outdated. I am waiting for my raspberry pi4 to be delivered and I hope to look into HA again and make it up to date. Could take a while tho..

**Comment by jampez77** [comment]
@SBRK  This weekend I updated my hassio to 0.98.1 and the @unknownskl repo appeared in the add on section. This version does not produce this issue and will connect to my Xbox fine. It is however struggling with controlling it.

**Comment by SBRK** [comment]
> @SBRK This weekend I updated my hassio to 0.98.1 and the @unknownskl repo appeared in the add on section. This version does not produce this issue and will connect to my Xbox fine. It is however struggling with controlling it.

Neat, I will have a try. By controlling it you mean sending controller inputs? Can you manage to turn it on/off? All I'd be interested in is the state and on/off commands

**Comment by jampez77** [comment]
@SBRK I have my Satellite TV connected through the Xbox. When I try to change channel (via Alexa) the command for changing to TV source is called instead of the channel number as was done in the past.

I will check it in a few hours after work as I have yet to add my username/password in the config yet.
